code/workspaces/web-app/src/components/stacks/StackCardActions.js

```diff
@@ -21,38 +22,47 @@ const styles = theme => ({
   },
 });
 
-const StackCardActions = ({ stack, openStack, deleteStack, editStack, userPermissions, openPermission,
-  deletePermission, editPermission, classes }) => (
-  <div className={classes.cardActions}>
-    {openStack && <PermissionWrapper className={classes.buttonWrapper} userPermissions={userPermissions} permission={openPermission}>
-      <PrimaryActionButton
-        disabled={!isReady(stack)}
-        onClick={() => openStack(stack)}
-        fullWidth
-      >
-        Open
-      </PrimaryActionButton>
-    </PermissionWrapper>}
-    {deleteStack && <PermissionWrapper className={classes.buttonWrapper} userPermissions={userPermissions} permission={deletePermission}>
-      <SecondaryActionButton
-        disabled={!isReady(stack)}
-        onClick={() => deleteStack(stack)}
-        fullWidth
-      >
-        Delete
-      </SecondaryActionButton>
-    </PermissionWrapper>}
-    {editStack && <PermissionWrapper className={classes.buttonWrapper} userPermissions={userPermissions} permission={editPermission}>
-      <SecondaryActionButton
-        disabled={!isReady(stack)}
-        onClick={() => editStack(stack)}
-        fullWidth
-      >
-        Edit
-      </SecondaryActionButton>
-    </PermissionWrapper>}
-  </div>
-);
+const StackCardActions = (props) => {
+  const currentUserId = useCurrentUserId();
+  return <PureStackCardActions currentUserId={currentUserId} {...props} />;
+};
+
+export const PureStackCardActions = ({ stack, openStack, deleteStack, editStack, userPermissions, openPermission,
+  deletePermission, editPermission, currentUserId, classes }) => {
+  const ownsNotebook = stack.users.includes(currentUserId);
+
+  return (
+    <div className={classes.cardActions}>
+      {openStack && <PermissionWrapper className={classes.buttonWrapper} userPermissions={userPermissions} permission={openPermission}>
+        <PrimaryActionButton
+          disabled={!isReady(stack)}
+          onClick={() => openStack(stack)}
+          fullWidth
+        >
+          Open
+        </PrimaryActionButton>
+      </PermissionWrapper>}
+      {deleteStack && ownsNotebook && <PermissionWrapper className={classes.buttonWrapper} userPermissions={userPermissions} permission={deletePermission}>
```

Comment:
Can't see a better way of doing it really but seems slightly odd to refer to notebooks specifically as this is applied for storage and sites too.

Reply:
That's a good point, this should definitely be `ownsStack` rather than `ownsNotebook`!